packages/grid/src/vaadin-grid-filter-element-mixin.js

```diff
@@ -70,7 +70,10 @@ export const GridFilterElementMixin = (superClass) =>
       this._filterController = new SlotController(this, '', 'vaadin-text-field', {
         initializer: (field) => {
           field.addEventListener('value-changed', (e) => {
-            if (field.__previousValue === undefined && e.detail.value === '') {
+            // In Lit version, two value-changed events occur during initialization: first,
+            // value set to undefined and then changes to empty string. Use `hasUpdated` to
+            // detect Lit as in Polymer there is only one event and this logic is not needed.
+            if (field.hasUpdated && field.__previousValue === undefined && e.detail.value === '') {
```

Comment:
Looks like the problem still exists for the Lit version when using `vaadin-grid-filter-column`:
```html
<script type="module">
  const grid = document.querySelector('vaadin-grid');
  grid.items = [
    { name: 'John', surname: 'Lennon' },
    { name: 'Paul', surname: 'McCartney' },
    { name: 'George', surname: 'Harrison' },
    { name: 'Ringo', surname: 'Starr' }
  ];
</script>
<vaadin-grid item-id-path="name">
  <vaadin-grid-filter-column path="name"></vaadin-grid-filter-column>
</vaadin-grid>
```
If you enter and then delete a filter it doesn't update the items.

Reply:
Would this be an alternative? Setting the filter value programmatically on the text field wouldn't work then, but you should probably set it on the filter element itself. Also requires updating a few tests to fire `input` events, assuming those test user input.

```diff
@@ -69,15 +69,8 @@
 
       this._filterController = new SlotController(this, '', 'vaadin-text-field', {
         initializer: (field) => {
-          field.addEventListener('value-changed', (e) => {
-            // In Lit version, two value-changed events occur during initialization: first,
-            // value set to undefined and then changes to empty string. Use `hasUpdated` to
-            // detect Lit as in Polymer there is only one event and this logic is not needed.
-            if (field.hasUpdated && field.__previousValue === undefined && e.detail.value === '') {
-              field.__previousValue = e.detail.value;
-              return;
-            }
-            this.value = e.detail.value;
+          field.addEventListener('input', (e) => {
+            this.value = e.target.value;
           });
 
           this._textField = field;
@@ -91,12 +84,8 @@
       if (path === undefined || value === undefined || !textField) {
         return;
       }
-      if (this._previousValue === undefined && value === '') {
-        return;
-      }
 
       textField.value = value;
-      this._previousValue = value;
 
       this._debouncerFilterChanged = Debouncer.debounce(this._debouncerFilterChanged, timeOut.after(200), () => {
         this.dispatchEvent(new CustomEvent('filter-changed', { bubbles: true }));

```

Reply:
Thanks, makes sense. Changed to use this logic and updated tests.